api/src/server.ts

```diff
@@ -55,25 +58,27 @@ class ApiServer {
       options: {
         connection: dbConnection,
         debug: this._debug,
-        config: this._config
+        config: this._config,
       },
       routes: {
         prefix: "/api/v0/",
       },
     });
 
-    this._server.route({ method: "GET", path: "/api/health", handler: () => "👍" });
+    this._server.route({
+      method: "GET",
+      path: "/api/health",
+      handler: () => "👍",
+    });
 
     this._server.route({
       method: "POST",
       path: "/api/log",
       handler: async (request, h) => {
-        const ipAddress =
-          request.headers["x-real-ip"] || request.info.remoteAddress;
         const payload = request.payload as LogEntryCreatePayload;
         try {
           await dbConnection.insertLogEntry({
-            ip_address: ipAddress,
+            session_id: payload.session_id,
```

Comment:
This is great suggestion, @aimichal! This sounds like a better pattern than what I was envisioning, so I will give a shot.

Reply:
@aimichal I wanted to run by you an idea for how I am thinking of implementing this. What I am thinking is:

* The first time a browser makes a request to the logging endpoint, the server checks for a cookie containing a session ID
* If it does not exist, then it creates a new one as a UUID
* It saves the log record with that UUID
* It then sets the browser session ID cookie using that UUID

The main problem with this approach is the log events can be made in rapid succession, so it might be that a user would be assigned multiple session IDs, if one request was not completed before another was submitted (though I don't know how likely that is).

Another approach would be to set up a cookie when a user first loads the page, though admittedly I am not sure how to do that, as the page is served up by nginx, rather than the Node.js code. Do you know if there is a way to set a cookie to a random value using nginx code?

Reply:
> What I am thinking is:

Yep, this sounds right!

> The main problem with this approach is the log events can be made in rapid succession, so it might be that a user would be assigned multiple session IDs, if one request was not completed before another was submitted (though I don't know how likely that is).

This is a good observation.

Could the initial request for the homepage could make a Javascript request to the logging endpoint to establish a session cookie which will (probably) complete before the user does anything?

Alternatively, you could introduce a "hello" API route that returns an HTTP 200 with no content, and then sneak it into the page with an `<img width=0 height=0 src="/api/hello">` which (as a side-effect) runs thru the cookie code.

> Do you know if there is a way to set a cookie to a random value using nginx code?

NGINX is incredibly powerful. It can be scripted with Lua in its configuration. It's kind of crazy, but powerful. So maybe it is! That said, I don't recommend it. :-)

In my opinion, you should own the cookie generation code in your application. It'll be right there where you read the cookie and it'll all be less surprising that way.

Reply:
Thank you for this clarification! I think having the initial request for the homepage make a JavaScript request to the logging endpoint is a great idea.

So, I updated the PR with code in the API that sets session IDs upon the first call to the logging endpoint.